### PR TITLE
unix: fix various memory leaks and undef behavior

### PR DESCRIPTION
**Pull request by indutny**
Kindly suggested by `cppcheck`.

/cc @saghul

### CODE REVIEW

File: src/unix/signal.c [hunk 1]
```diff
@@ -348,6 +348,7 @@ static void uv__signal_event(uv_loop_t* loop,
   int r;
 
   bytes = 0;
+  end = 0;
```
Comment (saghul) [inline src/unix/signal.c:351]: Is this one needed? How can end be accessed without being initialized in this function?

**Comment by saghul** [comment]
Minor comment about the 'end' variable, otherwise LGTM.

File: src/unix/signal.c (hunk 1 above)
Comment (indutny) [inline src/unix/signal.c:351]: From there: https://github.com/indutny/libuv/blob/6b1227638db48236b7aaf45777826f43c231ae41/src/unix/signal.c#L357

Comment (saghul) [inline src/unix/signal.c:351]: Ah, right! It's good to go then :-)

**Comment by indutny** [comment]
Thanks, landed in 646de34

**Pull request closed without merging**